### PR TITLE
Drain

### PR DESCRIPTION
**Pull request by tiptop96**
If anyone could give suggestions for the doc comments that would be greatly appreciated. 🥇

### CODE REVIEW

File: src/drain.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+}
+
+#[derive(StructOpt, Debug, Clone)]
+pub(crate) struct DrainCliCommand {
+    #[structopt(subcommand)]
+    selection: DrainSelection,
+}
+
+// Hack to get around the fact that we can't have a global 'output' arg.
```
Comment (brooksmtownsend) [inline src/drain.rs]: We should try and leave out `hack` from our code, maybe something like
```
Propagates output selection from CLI to all commands
```

File: src/main.rs [hunk 2]
```diff
@@ -38,6 +40,9 @@ struct Cli {
 
 #[derive(Debug, Clone, StructOpt)]
 enum CliCommand {
+    /// TODO
```
Comment (brooksmtownsend) [inline src/main.rs]: "Manage contents of local wasmCloud cache"

File: src/up.rs [hunk 3]
```diff
@@ -104,6 +105,10 @@ struct ReplCli {
 #[derive(StructOpt, Debug, Clone)]
 #[structopt(global_settings(&[AppSettings::ColorNever, AppSettings::DisableVersion, AppSettings::VersionlessSubcommands]))]
 enum ReplCliCommand {
+    // TODO
```
Comment (brooksmtownsend) [inline src/up.rs]: "Manage contents of local wasmCloud cache"

File: src/drain.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            DrainSelection::All(output)
+            | DrainSelection::Lib(output)
+            | DrainSelection::Oci(output) => output.kind,
+        }
+    }
+}
+
+#[derive(StructOpt, Debug, Clone)]
+pub(crate) enum DrainSelection {
+    /// TODO
```
Comment (brooksmtownsend) [inline src/drain.rs]: Remove all cached files created by wasmCloud

File: src/drain.rs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    All(Output),
+    /// TODO
```
Comment (brooksmtownsend) [inline src/drain.rs]: Remove cached files downloaded from OCI registries by wasmCloud

File: src/drain.rs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    Oci(Output),
+    /// TODO
```
Comment (brooksmtownsend) [inline src/drain.rs]: Remove cached binaries extracted from provider archives

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
One suggestion and I threw in what I thought appropriate doc comments would be!

File: src/drain.rs (hunk 1 above)
Comment (tiptop96) [reply src/drain.rs]: Yeah, that's a good point 👍

File: src/up.rs [hunk 7]
```diff
@@ -553,6 +567,12 @@ async fn handle_up(cmd: UpCliCommand) -> Result<()> {
     Ok(())
 }
 
+fn handle_drain(drain_cmd: DrainCliCommand, output_state: Arc<Mutex<OutputState>>) -> Result<()> {
+    let output = crate::drain::handle_command(drain_cmd)?;
+    log_to_output(output_state, output);
+    Ok(())
+}
```
Comment (brooksmtownsend) [inline src/up.rs:574]: Love how clean the functions are getting now 😄

**Review by brooksmtownsend — APPROVED** [review]
Looks great, and works great 😄

**Pull request merged**